core/templates/dev/head/pages/exploration-player-page/services/hints-and-solution-manager.service.ts

```diff
@@ -179,12 +179,9 @@ angular.module('oppia').factory('HintsAndSolutionManagerService', [
         }
 
         wrongAnswersSinceLastHintConsumed++;
-        if (!areAllHintsExhausted() && !isAHintWaitingToBeViewed()) {
+        if (!areAllHintsExhausted()) {
           if (numHintsReleased === 0 &&
-              wrongAnswersSinceLastHintConsumed >= 2) {
-            accelerateHintRelease();
-          } else if (
-            numHintsReleased > 0 && wrongAnswersSinceLastHintConsumed >= 1) {
```

Comment:
I've deleted this condition here because I checked out through the tests and if this condition is true then accelerateHintRelease won't work correctly (it will conflict with consumeHint function, releaseHint timeout to be specifically). I've changed this piece of code to readjust what I've deleted, so the condition above will consider both cases in a way that is not so restricted.

Reply:
I don't quite understand. Isn't `accelerateHintrelease` supposed to conflict with `releaseHint` so as to overwrite the timeout for that with the new one?
What was the exact issue that you faced here? Can you share repro steps, so I can check it? Since, I don't recall any issues here.

Reply:
During the tests, `releaseHint` was overwriting `accelerateHintrelease`. I can check it further and then I'll share with you

Reply:
Ok, thanks. It's just that I remember the accelerateHintsRelease working as expected some time back, not sure if it got changed recently.

Reply:
Hey @aks681 I took a look again in this service (before the changes I made) and now I think I can explain some paths I did during the tests:

1. I've removed `!isAHintWaitingToBeViewed` from this condition: 
` if (!areAllHintsExhausted() && !isAHintWaitingToBeViewed()) {`
Because the code will only evaluates this condition if the condition below is not true:
```
if (isAHintWaitingToBeViewed()) {
    return;
}
```
So evaluating `isAHintWaitingToBeViewed` and then `!isAHintWaitingToBeViewed` is redundant.

Here's is the path I made in order to know that the second condition:
`numHintsReleased > 0 && wrongAnswersSinceLastHintConsumed >= 1`
should be removed.

After `accelerateHintRelease` is called for the first time (and it would be the first condition), `isAHintWaitingToBeViewed` become true. Then `recordWrongAnswer` will always return before trying to call `accelerateHintRelease` again.
In order to make `isAHintWaitingToBeViewed` false, `displayHint` can be called. But then it will call `consumeHint`, increasing both `numHintsConsumed` and `numHintsReleased`, so this way is not correct, because `isAHintWaitingToBeViewed` will continue be true.
The last way I thought it's to call `displayHint` for all the hints and the call `recordWrongAnswer`, so `numHintsConsumed` and `numHintsReleased` would be equal. The problem here is that `areAllHintsExhausted` would be true and the conditions that would call `accelerateHintRelease` will never be evaluate.
So I've removed the second solution since it will never be true (according to the tests I did). But I'm not sure if it's 100% right. Is there other situation that I might forget?

Reply:
Also, I know that I said about the conflict in the first comment but not I'm not sure about that. This service is really tricky so the condition I've removed is either not working right or not even being evaluated.

Reply:
Let me add @kevintab95 to this thread, since he originally wrote the hints/solutions code and might be able to provide some clarification/context.

@kevintab95, would you mind reviewing the PR as well, just to make sure that everything is still correct and that we don't introduce any regressions?

Thanks!

Reply:
Oops, I totally missed this. Sorry!

Nice catch @marianazangrossi removing the redundant `!isAHintWaitingToBeViewed()`.

Hm, for the other condition can you please check this one scenario and see if it still works:

1. On landing on a card that has multiple hints, immediately enter 2 incorrect answers and wait for approx. 10s.
2. First hint should be available.
3. View this hint and immediately submit an incorrect answer.
4. Wait for approx 10s. You should see that a new hint is available.

I think if this scenario works there should not be any issues. Thanks!

Reply:
Hey @kevintab95 I'm sorry but I'm kinda lost trying to reproduce this scenario. The card you mentioned is in this image below?
![Captura de tela de 2020-02-22 02-35-35](https://user-images.githubusercontent.com/34922478/75087219-1bad2a00-551c-11ea-87e7-1c46b0760618.png)

Reply:
Hi @marianazangrossi, I'm unable to see the hints section in your screenshot (it should be in the bottom right of the screen). 

Btw, I checked out your branch and followed the steps mentioned above. I noticed that Step 4 does not work as expected. Here's the scenario:

After a hint is consumed and the user inputs an incorrect answer immediately, the next hint should be released faster (approx 10s) instead of the usual 30s wait time. This does not seem to work with your changes.

Please make sure this functionality works so that we don't introduce any regressions in this PR. Thanks!

Reply:
Hey @kevintab95 Thanks a lot for your help! I've fixed the unit tests, PTAL!